src/aggregator/client/writer_mgr.go

```diff
@@ -116,9 +116,9 @@ func newInstanceWriterManager(opts Options) (instanceWriterManager, error) {
 		doneCh:  make(chan struct{}),
 	}
 
-	pool, err := xsync.NewPooledWorkerPool(
-		opts.FlushWorkerCount(),
-		xsync.NewPooledWorkerPoolOptions().SetKillWorkerProbability(0.05),
+	pool, err := xsync.NewStaticPooledWorkerPool(xsync.NewPooledWorkerPoolOptions().
+		SetKillWorkerProbability(0.05).
```

Comment:
Hm do static pooled workers have ability to be killed? Thought they stuck around forever?

Reply:
They get killed and respawn. All the initial routines respawn and the dynamic ones do not